Userland/Services/WindowServer/Window.cpp

```diff
@@ -873,4 +873,21 @@ bool Window::is_descendant_of(Window& window) const
     return false;
 }
 
+bool Window::hit_test(const Gfx::IntPoint& point, bool include_frame) const
+{
+    if (!frame().rect().contains(point))
+        return false;
+    if (include_frame) {
+        if (!rect().contains(point))
+            return frame().hit_test(point);
+    } else if (!rect().contains(point)) {
+        return false;
+    }
```

Comment:
I think this would look better "inverted":

```cpp
if (!rect().contains(point)) {
    if (include_frame)
        return frame().hit_test(point);
    return false;
}
```